src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -1666,9 +2019,172 @@ public void fetchMetricsBasedOnProfileAndDatasource(KruizeObject kruizeObject, T
+                                String timestamp = sdf.format(new Date(epochTime * KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC));
+                                Date date = sdf.parse(timestamp);
+                                Timestamp dateTS = new Timestamp(date.getTime());
+                                interval_end_time_epoc = dateTS.getTime() / KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC
+                                        - ((long) dateTS.getTimezoneOffset() * KruizeConstants.TimeConv.NO_OF_SECONDS_PER_MINUTE);
+                                int maxDay = Terms.getMaxDays(kruizeObject.getTerms());
+                                LOGGER.info(KruizeConstants.APIMessages.MAX_DAY, maxDay);
+                                Timestamp startDateTS = Timestamp.valueOf(Objects.requireNonNull(dateTS).toLocalDateTime().minusDays(maxDay));
+                                interval_start_time_epoc = startDateTS.getTime() / KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC
+                                        - ((long) startDateTS.getTimezoneOffset() * KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC);
+                            }
+                        } else {
+                            // Convert timestamps to epoch time
+                            interval_end_time_epoc = interval_end_time.getTime() / KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC
+                                    - ((long) interval_end_time.getTimezoneOffset() * KruizeConstants.TimeConv.NO_OF_SECONDS_PER_MINUTE);
+                            interval_start_time_epoc = interval_start_time.getTime() / KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC
+                                    - ((long) interval_start_time.getTimezoneOffset() * KruizeConstants.TimeConv.NO_OF_MSECS_IN_SEC);
+                        }
+                        HashMap<Timestamp, IntervalResults> containerDataResults = new HashMap<>();
+                        IntervalResults intervalResults;
+                        HashMap<AnalyzerConstants.MetricName, MetricResults> resMap;
+                        HashMap<String, MetricResults> resultMap;
+                        MetricResults metricResults;
+                        MetricAggregationInfoResults metricAggregationInfoResults;
+
+                        List<Metric> metricList = metricProfile.getSloInfo().getFunctionVariables();
+
+                        // Iterate over metrics and aggregation functions
+                        for (Metric metricEntry : metricList) {
+                            HashMap<String, AggregationFunctions> aggregationFunctions = metricEntry.getAggregationFunctionsMap();
+                            for (Map.Entry<String, AggregationFunctions> aggregationFunctionsEntry: aggregationFunctions.entrySet()) {
+                                // Determine promQL query on metric type
+                                String metricQuery = aggregationFunctionsEntry.getValue().getQuery();
```

Comment:
No need of defining extra variable here, `promQL` variable can be assigned the query directly.

Reply:
updated the code to assign the query directly to the promQL variable. thanks for pointing out